Scripts/Runtime/Data/VirtualData/VirtualData.cs

```diff
@@ -0,0 +1,296 @@
+using Anvil.CSharp.Core;
+using Anvil.Unity.DOTS.Entities;
+using Anvil.Unity.DOTS.Jobs;
+using System;
+using System.Collections.Generic;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    /// <summary>
+    /// This class represents wrapped collections of data and manages them for use in Jobs.
+    /// </summary>
+    /// <remarks>
+    /// In Unity's ECS, data is stored on <see cref="Entity"/>'s via <see cref="IComponentBase"/> structs.
+    /// Unity's <see cref="SystemBase"/>'s handle the dependencies on the different sorts of data that are needed
+    /// for a given update call depending on the <see cref="EntityQuery"/>s used and Jobs scheduled.
+    ///
+    /// There are often cases where the overhead of using Entities doesn't make sense. The data is not persistent
+    /// and should exist for a period of time and then cease to exist. While this can be accomplished by Adding
```

Comment:
```suggestion
    /// There are often cases where the overhead of using Entities doesn't make sense. The data is not persistent
    /// and has a short lifecycle (<1-10 frames). While this can be accomplished by Adding
```

Reply:
I don't think that's accurate. 

A Timer could exist for 1 frame or 10 minutes.
Following a path could exist for a few seconds or a few minutes.

My point that I was trying to make is that it doesn't make sense to have it be on the Actor Entity because you won't ALWAYS have a path to follow or a timer going on. (Also in the timer case, you might want multiple timers so a component wouldn't work anyway). 

And you don't want to be a standalone entity that is somehow linked due to chunk thrashing as these overall collections blink in and out of existence.

Reply:
The decision on lifecyle length isn't about covering all cases just the most common one. If your timer's are usually 10mins then it probably does make more sense to use components on entities. Obviously we don't build the timer system to assume this since most timers are indeed short. The point of VirtualData is to remove the overhead of potentially short lived data right?

Or are you saying that Entity+Component should never have a structural change and the entity should exist permanently in a session. If any aspect of an entity is going to change throughout its lifetime it should be VD. 
If that's the case then what in a game would actually be Entity + Component? Nearly everything has a lifecycle. 

Should a crew member be totally VirtualData and not bound to an entity since they can die?

I think what I'm getting at is that this documentation needs to specify when VD makes sense. Reading "should exist for a period of time and then cease to exist" doesn't tell me anything as a developer. Everything in an application meets that definition.

> And you don't want to be a standalone entity that is somehow linked due to chunk thrashing as these overall collections blink in and out of existence.

If it's not a short lifecycle what's the issue with thrashing the chunk? It's not like it will become fragmented if you have a reasonable amount of churn.

Reply:
> The point of VirtualData is to remove the overhead of potentially short lived data right?

Somewhat. You should use VirtualData when the data is shortlived or when you have multiple different "completion" avenues or when you want to avoid large entities etc.

> Or are you saying that Entity+Component should never have a structural change and the entity should exist permanently in a session. If any aspect of an entity is going to change throughout its lifetime it should be VD.
> If that's the case then what in a game would actually be Entity + Component? Nearly everything has a lifecycle.
> 
> Should a crew member be totally VirtualData and not bound to an entity since they can die?

In my mind, I would use Entity+Component to represent an Actor and it's corresponding render information because that's what the URP is expecting. 

I'm not sure I see a benefit to using it for anything else right now. Maybe the DynamicBuffer consolidation of Attributes. 

> I think what I'm getting at is that this documentation needs to specify when VD makes sense. Reading "should exist for a period of time and then cease to exist" doesn't tell me anything as a developer. Everything in an application meets that definition.

Sure, ok. I'll try and reword a bit. In essence, use `VirtualData` first unless it is easier to use Entity+Components for your specific use case. I would argue that most cases are not.

> If it's not a short lifecycle what's the issue with thrashing the chunk? It's not like it will become fragmented if you have a reasonable amount of churn.

I observed a ton of chunk thrashing when the lifespan was variable. Take timers and following paths for example. 
When both were Entity+Component some timers are short (1.0 seconds) and some are longer (3.0 seconds). Some paths are a few tiles away, some are the whole map away. 

I would have say 10 chunks of these entities and on frame 1 all of them are full which makes sense. Then as Timers complete or paths complete, pieces of the chunks will fall out. 

What you would expect is that as new ones come into existence it would fill up the empty space in the first chunk and then fill up the second chunk etc and eventually reach a point where the longest lasting Timers or Paths in the 10th chunk are all gone and on average we have let's say 6 chunks in flight. And that last chunk would ebb and flow as things came in and out of existence. 

But in practice that's not what happened. All 10 chunks remained and all 10 ebbed and flowed throughout execution. I'm not sure how Unity is handling the recycling of Entities but it seemed to keep the chunks open and have the churn.

Reply:
> > The point of VirtualData is to remove the overhead of potentially short lived data right?
> 
> Somewhat. You should use VirtualData when the data is shortlived or when you have multiple different "completion" avenues or when you want to avoid large entities etc.

OK, the docs need to define that as well as what you mean by "shortlived"


> I observed a ton of chunk thrashing when the lifespan was variable. Take timers and following paths for example.

That's strange because the forum posts seem to indicate that chunks will have elements backfill their gaps.

Reply:
> OK, the docs need to define that as well as what you mean by "shortlived"

Yep docs are updated.

> That's strange because the forum posts seem to indicate that chunks will have elements backfill their gaps.

This is what I thought too.